liana-gui/src/launcher/launcher_update.rs

```diff
@@ -0,0 +1,368 @@
+        })
+    }
+
+    fn do_create_wallet(&mut self) -> Task<Message> {
+        let datadir_path = self.datadir_path.clone();
+        let network = self.network;
+        Task::perform(async move { (datadir_path, network) }, |(d, n)| {
+            Message::Install(d, n, UserFlow::CreateWallet)
+        })
+    }
+
+    fn do_share_xpubs(&mut self) -> Task<Message> {
+        let datadir_path = self.datadir_path.clone();
+        let network = self.network;
+        Task::perform(async move { (datadir_path, network) }, |(d, n)| {
+            Message::Install(d, n, UserFlow::ShareXpubs)
+        })
+    }
+
+    fn do_delete_wallet(&mut self) -> Task<Message> {
+        let wallet_datadir = self.datadir_path.join(self.network.to_string());
+        let config_path = wallet_datadir.join(app::config::DEFAULT_FILE_NAME);
+        let internal_bitcoind = if let Ok(cfg) = app::Config::from_file(&config_path) {
+            Some(cfg.start_internal_bitcoind)
+        } else {
+            None
+        };
+        self.delete_wallet_modal = Some(DeleteWalletModal::new(
+            self.network,
+            wallet_datadir,
+            internal_bitcoind,
+        ));
+        Task::none()
+    }
+
+    fn do_select_network(&mut self, network: Network) -> Task<Message> {
+        self.network = network;
+        Task::perform(
+            check_network_datadir(self.datadir_path.clone(), self.network),
+            Message::Checked,
+        )
+    }
+
+    fn do_run(&mut self) -> Task<Message> {
+        if matches!(self.state, WalletState::Wallet { .. }) {
+            let datadir_path = self.datadir_path.clone();
+            let mut path = self.datadir_path.clone();
+            path.push(self.network.to_string());
+            path.push(app::config::DEFAULT_FILE_NAME);
+            let cfg = app::Config::from_file(&path).expect("Already checked");
+            let network = self.network;
+            Task::perform(async move { (datadir_path.clone(), cfg, network) }, |m| {
+                Message::Run(m.0, m.1, m.2)
+            })
+        } else {
+            Task::none()
+        }
+    }
+
+    fn default_update(&mut self, message: &Message) -> Task<Message> {
+        if let Some(modal) = &mut self.delete_wallet_modal {
+            return modal.update(message.clone());
+        }
+        Task::none()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Message {
+    View(ViewMessage),
+    Install(PathBuf, Network, UserFlow),
+    Checked(Result<WalletState, String>),
+    Run(PathBuf, app::config::Config, Network),
+}
+
+#[derive(Debug, Clone)]
+pub enum ViewMessage {
+    ImportWallet,
+    CreateWallet,
+    ShareXpubs,
+    SelectNetwork(Network),
+    StartInstall(Network),
+    Check,
+    Run,
+    DeleteWallet(DeleteWalletMessage),
+}
+
+#[derive(Debug, Clone)]
+pub enum DeleteWalletMessage {
+    ShowModal,
+    CloseModal,
+    Confirm,
+    Deleted,
+}
+
+pub(super) struct DeleteWalletModal {
```

Comment:
DeleteWalletModal can be split out too, into `children` module of `launcher` module.